sa/sa.go

```diff
@@ -318,30 +321,37 @@ func (ssa *SQLStorageAuthority) CountCertificatesByNames(ctx context.Context, re
 					return
 				default:
 				}
-				currentCount, err := ssa.countCertificatesByName(ssa.dbReadOnlyMap.WithContext(ctx), domain, req.Range)
+				count, earliest, err := ssa.countCertificatesByName(ssa.dbReadOnlyMap.WithContext(ctx), domain, req.Range)
 				if err != nil {
 					results <- result{err: err}
 					// Skip any further work
 					cancel()
 					return
 				}
 				results <- result{
-					count:  currentCount,
-					domain: domain,
+					count:    count,
+					earliest: earliest,
+					domain:   domain,
 				}
 			}
 		}()
 	}
 	wg.Wait()
 	close(results)
+	// Set earliest to the latest possible time, so that we can find the
+	// earliest certificate in the results.
+	earliest := time.Unix(0, req.Range.Latest)
 	counts := make(map[string]int64)
 	for r := range results {
 		if r.err != nil {
 			return nil, r.err
 		}
 		counts[r.domain] = r.count
+		if r.earliest != nil && earliest.After(*r.earliest) {
```

Comment:
I'd reverse this one too
```suggestion
		if r.earliest != nil && *r.earliest.Before(earliest) {
```